supervision/tracker/byte_tracker/byte_tracker.py

```diff
@@ -0,0 +1,483 @@
+            track_buffer (int, optional): Number of frames to buffer when a track is lost.
+            match_thresh (float, optional): Threshold for matching tracks with detections.
+            aspect_ratio_thresh (float, optional): Threshold for aspect ratio to filter out detections.
+            min_box_area (float, optional): Minimum box area threshold to filter out detections.
+            mot20 (bool, optional): Set to True for using MOT20 evaluation criteria.
+            frame_rate (int, optional): The frame rate of the video.
+
+        Attributes:
+            track_thresh (float): Detection confidence threshold for track activation.
+            track_buffer (int): Number of frames to buffer when a track is lost.
+            match_thresh (float): Threshold for matching tracks with detections.
+            aspect_ratio_thresh (float): Threshold for aspect ratio to filter out detections.
+            min_box_area (float): Minimum box area threshold to filter out detections.
+            mot20 (bool): Indicates if MOT20 evaluation criteria are used.
+            tracked_stracks (List[strack]): List of tracked strack objects.
+            lost_stracks (List[strack]): List of lost strack objects.
+            removed_stracks (List[strack]): List of removed strack objects.
+            frame_id (int): ID of the current frame.
+            det_thresh (float): Detection threshold for track association.
+            buffer_size (int): Size of the buffer for a track when it's lost.
+            max_time_lost (int): Maximum number of frames a track can be lost before removal.
+            kalman_filter (KalmanFilter): The Kalman filter used for state prediction.
+        """
+        self.track_thresh = track_thresh
+        self.track_buffer = track_buffer
+        self.match_thresh = match_thresh
+        self.aspect_ratio_thresh = aspect_ratio_thresh
+        self.min_box_area = min_box_area
+        self.mot20 = mot20
+
+        self.tracked_stracks = []  # type: list[strack]
+        self.lost_stracks = []  # type: list[strack]
+        self.removed_stracks = []  # type: list[strack]
+
+        self.frame_id = 0
+        self.det_thresh = self.track_thresh + 0.1
+        self.buffer_size = int(frame_rate / 30.0 * self.track_buffer)
+        self.max_time_lost = self.buffer_size
+        self.kalman_filter = KalmanFilter()
+
+    def update_from_detections(self, detections, img_info, img_size):
+        """
+        Update a matched strack. It uses the supervision detection.
+
+        Parameters:
+            detections: The new detections to update with.
+            img_info: Image Info.
+            img_size: Image Size
+
+        Updates the strack with the provided results and frame info.
+        Returns:
+            Detection: supervision detection result with track id
+        """
+        # update tracker
+        tracks = self.update_from_numpy(
```

Comment:
@dbroboflow Just try to convert `tracks` to `sv.Detections` format and you should be fine.

Reply:
Hi Hardik,

We will use it in the inference server. Because of that, there are separated methods.